### PR TITLE
fix(treeview): fixed supplier, item, customer, sales person and territory group for treeview

### PR DESCRIPTION
**Pull request by vishdha**
TASK ID: [From the List view, I am not able to create a child node in their Parent Node.](https://app.asana.com/0/1192407894480744/1198506200773421)

Depends on : https://github.com/Bloomstack/frappe/pull/318
before:
Peek 2020-11-03 17-23

after:
Peek 2020-11-03 17-21

### CODE REVIEW

File: erpnext/setup/doctype/territory/territory.js [hunk 1]
```diff
@@ -11,29 +11,23 @@ frappe.ui.form.on("Territory", {
...
+					['Territory', 'is_group', '=', 1],
+					['Territory', 'name', "!=", doc.territory_name]
+				]
+			}
+		}
+	},
+	refresh: function(frm) {
+		frm.trigger("set_root_readonly");
+	},
+	set_root_readonly: function(frm) {
```
Comment (neilLasrado) [inline erpnext/setup/doctype/territory/territory.js]: @vishdha we must make this a common function in utils maybe, the code seems repetitive.

File: erpnext/setup/doctype/customer_group/customer_group.js [hunk 2]
```diff
@@ -1,37 +1,29 @@
...
-			'is_group': 1
+frappe.ui.form.on('Customer Group', {
+	setup: function(frm) {
+		frm.fields_dict['parent_customer_group'].get_query = function(doc) {
+			return {
+				filters:[
+					['Customer Group', 'is_group', '=', 1],
+					['Customer group', 'name', "!=", doc.customer_group_name]
+				]
+			}
```
Comment (hrwX) [inline erpnext/setup/doctype/customer_group/customer_group.js]:
```suggestion
		frm.set_query('parent_customer_group', function(doc) {
			return {
				filters: {
					'is_group': 1,
					'name': ["!=", doc.customer_group_name]
				}
			}
		});
```

Do these changes for every query on the link field

File: erpnext/setup/doctype/territory/territory.js [hunk 3]
```diff
@@ -11,29 +11,17 @@ frappe.ui.form.on("Territory", {
...
+				filters:[
+					['Territory', 'is_group', '=', 1],
+					['Territory', 'name', "!=", doc.territory_name]
+				]
+			}
+		}
+	},
+	refresh: function(frm) {
+		// read-only for root territory group
+		frm.set_root_read_only("parent_territory");
```
Comment (hrwX) [inline erpnext/setup/doctype/territory/territory.js]: - This doesn't seem like a fix for the Treeview, only code is being moved.
- Ask for the roles assigned to users who are facing this issue.
- This issue cannot be replicated with administrator access.

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/setup/doctype/territory/territory.js (hunk 3 above)
Comment (vishdha) [reply erpnext/setup/doctype/territory/territory.js]: @hrwX  its fix for the task [From the List view, I am not able to create a child node in their Parent Node.](https://app.asana.com/0/1192407894480744/1198506200773421)

Where when users try to create a child or root node from the list view.
Had discussed with Swati facing this issue for all role when they try to create a doc from list view

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**